test/controllers/todos_controller_test.rb

```diff
@@ -1,7 +1,69 @@
 require "test_helper"
 
 class TodosControllerTest < ActionDispatch::IntegrationTest
-  # test "the truth" do
-  #   assert true
-  # end
+  @created_todo
+  setup do
+    @valid_params={
+      title: "Valid Title",
+      status: "created",
+      is_completed: false
+    }
+    @invalid_params={
+      title: "title for invalid",
+      is_completed: false
+    }
+    @todo = Todo.create!(@valid_params)
+  end
+
+  test "#index should return response :ok" do
+    get todos_url
+    assert_response :ok
+  end
+
+  test "#index should return an Array" do
+    get todos_url
+    body = JSON.parse(response.body)
+    assert_kind_of(Array, body)
+  end
+
+  test "#show should respond with :ok when fetched with valid Todo id" do
+    get todo_url(@todo)
+    assert_response :ok
+  end
+
+  test "#show fetched Todo should match created Todo" do
+    get todo_url(@todo)
+    fetched_todo = JSON.parse(response.body)
+
+    assert_equal @todo.title, fetched_todo["title"]
+    assert_equal @todo.status, fetched_todo["status"]
+    assert_equal @todo.is_completed, fetched_todo["is_completed"]
+  end
+
+  test "#create should create todo with valid params" do
+    assert_difference("Todo.count") do
+      post todos_url, params: { todo: @valid_params }
+    end
+  end
+
+  test "#create should return a Hash" do
```

Comment:
It isn't just returning a Hash but the Todo that was meant to be created. I might validate that instead of just the type.